nilearn/connectome/connectivity_matrices.py

```diff
@@ -390,4 +471,49 @@ def transform(self, X):
                                  '"covariance" and "precision", got kind '
                                  '"{}"'.format(self.kind))
 
-        return np.array(connectivities)
+        connectivities = np.array(connectivities)
+        if self.kind != 'tangent':
+            self.mean_ = connectivities.mean(axis=0)
+
+        if self.vectorize:
+            connectivities = sym_matrix_to_vec(connectivities)
+
+        return connectivities
+
+    def _check_fitted(self):
+        if not hasattr(self, "cov_estimator_"):
+            raise ValueError('It seems that {0} has not been fitted. '
+                             'You must call fit() before calling '
+                             'transform().'.format(self.__class__.__name__)
+                             )
+
+    def inverse_transform(self, connectivities):
+        """Returns connectivity matrices from vectorized or not connectivities.
+        If kind is 'tangent', the covariance matrices are reconstructed.
+
+        Parameters
+        ----------
+        connectivities : list of n_subjects numpy.ndarray with shapes\
+            (n_features, n_features) or (n_features * (n_features + 1) /2,)
+            Connectivities of each subject, vectorized or not.
+
+        Returns
+        -------
+        output : numpy.ndarray, shape (n_subjects, n_features, n_features)
+             The corresponding connectivity matrices.
+             If kind is 'tangent', the covariance matrices are reconstructed.
+        """
+        self._check_fitted()
+
+        connectivities = np.array(connectivities)
+        if self.vectorize:
```

Comment:
>I didn't understand why converting to symmetric matrices is linked to `self.vectorize`

because we only vectorize if `self.vectorize=True`, otherwise we already have matrices, no ?